src/value-types.js

```diff
@@ -0,0 +1,218 @@
+var execute = require('./execute');
+
+var Value = function Value(value) {
+    this.value = value;
+};
+
+Value.prototype.equals = function(b) {
```

Comment:
I'm following that pattern these days as well. Crockford is my inspiration - he says he doesn't use this anymore.